### ISSUE TITLE
SendPingAsyncWithIPAddress fails with "The operation completed successfully."

### ISSUE THREAD

**Issue by jkotas** [dotnet/corefx#35351]
SendPingAsyncWithIPAddress  failed in https://github.com/dotnet/coreclr/pull/22619:

```
System.Net.NetworkInformation.Tests.PingTest.SendPingAsyncWithIPAddress(addressFamily: InterNetwork) [FAIL]
  System.Net.NetworkInformation.PingException : An exception occurred during a Ping request.
  ---- System.ComponentModel.Win32Exception : The operation completed successfully.
  Stack Trace:
     D:\j\workspace\windows-TGrou---74aa877a\src\System.Net.Ping\tests\FunctionalTests\PingTest.cs(749,0): at System.Net.NetworkInformation.Tests.PingTest.SendPingAsync(Func`2 sendPing, Action`1 pingResultValidator)
     D:\j\workspace\windows-TGrou---74aa877a\src\System.Net.Ping\tests\FunctionalTests\PingTest.cs(131,0): at System.Net.NetworkInformation.Tests.PingTest.SendPingAsyncWithIPAddress(AddressFamily addressFamily)
     --- End of stack trace from previous location where exception was thrown ---
     ----- Inner Stack Trace -----
     F:\vsagent\13\s\src\System.Net.Ping\src\System\Net\NetworkInformation\Ping.Windows.cs(96,0): at System.Net.NetworkInformation.Ping.DoSendPingCore(IPAddress address, Byte[] buffer, Int32 timeout, PingOptions options, Boolean isAsync)
     F:\vsagent\13\s\src\System.Net.Ping\src\System\Net\NetworkInformation\Ping.Windows.cs(42,0): at System.Net.NetworkInformation.Ping.SendPingAsyncCore(IPAddress address, Byte[] buffer, Int32 timeout, PingOptions options)
     F:\vsagent\13\s\src\System.Net.Ping\src\System\Net\NetworkInformation\Ping.cs(331,0): at System.Net.NetworkInformation.Ping.SendPingAsync(IPAddress address, Int32 timeout, Byte[] buffer, PingOptions options)
```

**Comment by jkotas** [comment]
The test failure itself may be caused by intermittent network issue.

However, the error message "The operation completed successfully." is wrong. It suggests that the error handling is not right. We should be reporting the actual error.

**Comment by davidsh** [comment]
@wf

**Comment by davidsh** [comment]
@wfurt

**Comment by wfurt** [comment]
I have PR refactoring PING error handling but I did limited testing on windows. It is more difficult to inject errors there as we use OS API.

**Comment by wfurt** [comment]
fixed by #33621 and dotnet/coreclr#22871
